src/core/events/bpf.rs

```diff
@@ -284,18 +325,32 @@ where
     parse_raw_section::<T>(&raw_sections.pop().unwrap())
 }
 
+#[derive(Debug, Default, serde::Deserialize, serde::Serialize)]
+pub(crate) struct TaskEvent {
+    /// Process id.
+    pub(crate) pid: i32,
+    /// Thread group id.
+    pub(crate) tgid: i32,
+    /// Name of the current task.
+    pub(crate) comm: String,
+}
+
 #[event_section]
-#[repr(C)]
 pub(crate) struct CommonEvent {
     /// Timestamp of when the event was generated.
     pub(crate) timestamp: u64,
+    pub(crate) task: Option<TaskEvent>,
 }
 
-unsafe impl Plain for CommonEvent {}
-
 impl EventFmt for CommonEvent {
     fn event_fmt(&self, f: &mut fmt::Formatter, _: DisplayFormat) -> fmt::Result {
-        write!(f, "{}", self.timestamp)
+        write!(f, "{}", self.timestamp)?;
+
+        if let Some(current) = &self.task {
+            write!(f, " [{}] {}/{}", current.comm, current.tgid, current.pid)?;
+        }
```

Comment:
about moving, I guess it's a matter of taste and could make sense. We don't necessarily need to group things by ownership.
About pid/tgid, IDK, even if the same, it will make it more obvious that the event got triggered by the main thread. I have no strong opinion on this, I'm fine with both.

Reply:
I agree, but not for the same reasons :-D.

> It's a little bit less contained but the CommonEvent is already a special case. It would put event related info (ts, type of probe, probe) first and then event data (all other info / sections).

IMHO, those are internal implementation details that should not be exposed (when possible) to  users. When looking at what output makes more sense I would look it 100% from a user's pov.

Having said that, I think putting the comm (and in the future the cpu_id?) at the beginning makes more sense semantically: timestamp and comm are generic metadata, where as the probe's symbols is the first (and probably one of the most imporant) pieces of data. For me it makes sense that way "[metadata] [event core info] [event-specific data]"

Reply:
I turned the pid to optional (if pid != tgid) as you suggested.